#include "helpwindow.h"
#include "ui_helpwindow.h"

HelpWindow::HelpWindow(QWidget *parent) :
    QMainWindow(parent),
    ui(new Ui::HelpWindow)
{
    ui->setupUi(this);

    changeTextGame();

    connect(ui->gameButton,SIGNAL(clicked(bool)),SLOT(changeTextGame()));
    connect(ui->howButton,SIGNAL(clicked(bool)),SLOT(changeTextHow()));
    connect(ui->resultButton,SIGNAL(clicked(bool)),SLOT(changeTextResult()));
    connect(ui->exitButton,SIGNAL(clicked(bool)),SLOT(close()));
}

HelpWindow::~HelpWindow()
{
    delete ui;
}

void HelpWindow::changeTextGame()
{
    ui->textBrowser->setText(QString("Курсовая работа.\n    Вариант № 22 Игра на поиск предметов в картинке.\n\n"
                                     "  В данной игре вам предстоит проявить свою внимательность, чтобы "
                                     "найти все предметы на картинке. У вас будет изображение искомого предмета, "
                                     "но не спешите расслабляться, ведь все предметы будут хорошо спрятаны, а вы "
                                     "ограничены во времени. Будьте аккуратны ведь у вас только 3 попытки!\n"
                                     "  Бейте свои рекорды, улучшайте свои навыки и просто проводите весело время в игре!\n\n"
                                     "Автор: Бессонов Александр ИП-014."));
    ui->textBrowser->setAlignment(Qt::AlignJustify);
    ui->gameButton->setDefault(true);
    ui->howButton->setDefault(false);
    ui->resultButton->setDefault(false);
}

void HelpWindow::changeTextHow()
{
    ui->textBrowser->setText(QString("  Для того, что бы начать играть нажмите на кнопку \"Играть\" в главном меню игры.\n"
                                     "  В окне игры слева картинка, на которой надо будет находить предметы. На правой же стороне "
                                     "есть: текущее количество очков, счетчик предметов, изображение искомого предмета, "
                                     "время на поиск этого предмета, количество жизней, а так же 2 кнопки: \"Закончить игру\" и "
                                     "\"Выйти в меню\".\n"
                                     "  Вам надо нажимать на искомый предмет на картинке слева, если вы нажмете не туда у вас "
                                     "отнимится 1 жизнь. На поиск одного предмета дается 90 секунд, если вы не успеете найти "
                                     "предмет у вас отнимится 1 жизнь.\n"
                                     "  Изначально у вас есть 3 жизни, после каждой потери жизни предмет меняется и время на "
                                     "поиск обнуляется. Если у вас закончатся жизни, то вы проиграете.\n"
                                     "  Если вы нажимаете на искомый предмет, то оставшееся время сохраняется и суммируется "
                                     "с предыдущим (если такое было). После чего расчитывается текущий счет, количество очков = "
                                     "сумма сохраненного времени * (1 + количество жизней).\n"
                                     "  Если вы найдете все предметы или у вас закончатся жизни, игра закончится и вы сможете "
                                     "записать свой результат, подробнее об этом в разделе \"Результаты\"."));
    ui->textBrowser->setAlignment(Qt::AlignJustify);
    ui->gameButton->setDefault(false);
    ui->howButton->setDefault(true);
    ui->resultButton->setDefault(false);
}

void HelpWindow::changeTextResult()
{
    ui->textBrowser->setText(QString("  После игры у вас появится возможность записать свой результат в базу данных. Для этого "
                                     "поставьте галочку в ячейке рядом с \"Добавить мой результат в таблицу\", после этого "
                                     "вам обязательно надо указать имя. Если же вы не хотите добавлять свой результат, то уберите галочку.\n"
                                     "  В главном меню есть кнопка \"Результаты\", там вы сможете посмотреть все записанные результы. Данная "
                                     "таблица имеет три столбца: \"Номер результата\", \"Имя пользователя\" и \"Количество очков\". Все строки "
                                     "отсортированы по количеству очков.\n"
                                     "  Каждый результат имеет свой уникальный номер, по нему можно узнать насколько стара запись, так как "
                                     "нумерация идет от 1 по возрастанию.\n"
                                     "  База данных с результатами находится в папке с приложением и имеет название \"results.db\"."));
    ui->textBrowser->setAlignment(Qt::AlignJustify);
    ui->gameButton->setDefault(false);
    ui->howButton->setDefault(false);
    ui->resultButton->setDefault(true);
}
